src/vm/corhost.cpp

```diff
@@ -1220,7 +1210,6 @@ STDMETHODIMP CorHost2::UnloadAppDomain2(DWORD dwDomainId, BOOL fWaitUntilDone, i
         if (1 == refCount)
         {
             // Stop coreclr on unload.
-            m_fStarted = FALSE;
```

Comment:
Shouldn't we keep this one. If I cause the entire EE to shutdown by calling `UnloadAppDomain`, then I should get a failure from calling `Stop` after that.
We can't rely on the global I assume, since the `g_fEEStarted` is never reset.

Reply:
`Stop()` shouldn't cause a failure if the appDomain has been unloaded as that is the normal calling sequence for a given thread.  Across multiple threads it is ineffective as it is a member variable of a typically per thread object.

`g_fForbidEnterEE` is atomically set to true at an appropriate time by the `EEShutDown()`.  It is also checked by `CanRunManagedCode()` which is called by `IsRuntimeActive()`.

Reply:
I meant this for the "second" host - you're right that for the first host which actually starts the CLR this already works and your change has no effect there (in this particular case). With your change we're now going to take a different codepath through Stop for the second host... interestingly enough they both return the same failure code, so I guess it doesn't really matter. Feel free to ignore.